src/common/settings.js

```diff
@@ -58,32 +58,31 @@ const merges =
 
 export const fetch =
   (names/*:Array<Name>*/)/*:Task<Never, Result<Error, Settings>>*/ =>
-  Task.future(() => {
+  Task.create(resolve => {
     if (navigator.mozSettings != null) {
       const lock = navigator.mozSettings.createLock();
       const settings = names.map(name => lock.get(name));
-
-      return Promise.all(settings)
-                    .then(merges)
-                    .then(ok, error);
+      Promise.all(settings)
+        .then(merges)
+        .then(ok, error)
+        .then(resolve);
     } else {
-      return Promise.resolve(error(NotSupported));
+      resolve(error(NotSupported));
     }
   });
 
 
 export const change =
   (settings/*:Settings*/)/*:Task<Never, Result<Error, Settings>>*/ =>
-  Task.future(() => {
+  Task.create(resolve => {
     if (navigator.mozSettings != null) {
       const lock = navigator.mozSettings.createLock();
-      const result =
-        lock
+      lock
         .set(settings)
-        .then(always(ok(settings)), error);
-      return result
+        .then(always(ok(settings)), error)
+        .then(resolve);
     } else {
-      return Promise.resolve(error(NotSupported));
+      resolve(error(NotSupported));
```

Comment:
I'm afraid we need to keep the Promise.resolve here in order to preserve async resolution as prior to this change (See https://github.com/browserhtml/browserhtml/pull/1003#issuecomment-211648650 for more details)

```